ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Ledger/Basics.hs

```diff
@@ -165,6 +169,17 @@ class ( -- Requirements on the ledger state itself
 applyChainTick :: IsLedger l => LedgerCfg l -> SlotNo -> l -> Ticked l
 applyChainTick = lrResult ..: applyChainTickLedgerResult
 
+
+-- | Handler for ledger events
+newtype LedgerEventHandler m l =
```

Comment:
This very much seems like a `Tracer`. Would you object to using the `Tracer` interface for it?

Reply:
Is `l` somehow more convenient in your use case than the more primary `blk`? EG you could have `blk -> AuxLedgerEvent (LedgerState blk) -> m ()` (or maybe it's `ExtLedgerState`).

Reply:
`Tracer` conveys the wrong semantic, doesn't it? I agree that the interface is similar but they are used for vastly different things (although you could simple trace events as well, true). 

Regarding the `l` parameter, I recall we had to stick to `l` because some functions in the consensus are _really abstract_ and don't even have `blk` in context.

Reply:
> Is l somehow more convenient in your use case than the more primary blk?

If we replace `l` with `blk`, then the line `(lrEvents result)` won't compile because it needs the type parameter `l` and not `blk`.